src/server/controllers/user.js

```diff
@@ -9,6 +9,12 @@ const errorMessage = 'Invalid username or password.'
  const registerUser = async (req, res) => {
     const { username, password } = req.body;
 
+    if(!username || !password) {
+        res.status(401)
+        res.json({error: "All fields required"})
+        return
+    }
+
     const hashedPassword = bcrypt.hashSync(password, 10)
```

Comment:
Nice use of hashSync to get your required hash synchronously rather than returning the promise.

Reply:
Tbh I only used it becuase thats what seemed to work last time. I like the way you did it more (with await) but didnt want to mess it up by refactoring ha